NitroxClient/GameLogic/MobileVehicleBay.cs

```diff
@@ -22,36 +20,22 @@ public MobileVehicleBay(IPacketSender packetSender, Vehicles vehicles)
             this.vehicles = vehicles;
         }
 
-        public void BeginCrafting(GameObject constructor, TechType techType, float duration)
+        public void BeginCrafting(ConstructorInput constructor, GameObject constructedObject, TechType techType, float duration)
         {
-            NitroxId constructorId = NitroxEntity.GetId(constructor);
+            NitroxId constructorId = NitroxEntity.GetId(constructor.gameObject);
 
             Log.Debug($"Building item from constructor with id: {constructorId}");
 
-            Optional<object> opConstructedObject = TransientLocalObjectManager.Get(TransientObjectType.CONSTRUCTOR_INPUT_CRAFTED_GAMEOBJECT);
+            VehicleModel vehicleModel = vehicles.BuildVehicleModelFrom(constructedObject, techType);
+            vehicles.AddVehicle(vehicleModel);
 
-            if (opConstructedObject.HasValue)
-            {
-                GameObject constructedObject = (GameObject)opConstructedObject.Value;
-                List<InteractiveChildObjectIdentifier> childIdentifiers = VehicleChildObjectIdentifierHelper.ExtractInteractiveChildren(constructedObject);
-
-                VehicleModel vehicleModel = vehicles.BuildVehicleModelFrom(constructedObject, techType);
-                vehicles.AddVehicle(vehicleModel);
-
-                packetSender.Send(new ConstructorBeginCrafting(vehicleModel, constructorId, duration));
+            packetSender.Send(new ConstructorBeginCrafting(vehicleModel, constructorId, duration));
 
-                vehicles.SpawnDefaultBatteries(constructedObject, childIdentifiers);
-
-                MonoBehaviour monoBehaviour = constructor.GetComponent<MonoBehaviour>();
-                //We want to store the fallen position of the object to avoid flying object on reload 
-                if (monoBehaviour)
-                {
-                    monoBehaviour.StartCoroutine(vehicles.UpdateVehiclePositionAfterSpawn(vehicleModel, constructedObject, duration + 10.0f));
-                }
-            }
-            else
+            MonoBehaviour monoBehaviour = constructor.GetComponent<MonoBehaviour>();
+            //We want to store the fallen position of the object to avoid flying object on reload 
+            if (monoBehaviour)
             {
-                Log.Error("Could not send packet because there wasn't a corresponding constructed object!");
+                monoBehaviour.StartCoroutine(vehicles.UpdateVehiclePositionAfterSpawn(vehicleModel, constructedObject, duration + 10.0f));
```

Comment:
We should definetely find something more sexier than that. I added that as a temporary workaround for people

Reply:
There is a good chance this my get solved when I move vehicles over to the entity system.  We'll have to see.